### ISSUE TITLE
Repo: packages/*/generated gets picked up by linting, but shouldn't

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#10123]
### Suggestion

```shell
git checkout main
yarn build
yarn lint
```

```plaintext
✖  nx run repo:lint
      > eslint . --ignore-pattern=packages --cache
      
      
      /Users/josh/repos/typescript-eslint/docs/packages/ast-spec/generated/typedoc-sidebar.cjs
        0:0  error  Parsing error: /Users/josh/repos/typescript-eslint/docs/packages/ast-spec/generated/typedoc-sidebar.cjs was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject
      
      /Users/josh/repos/typescript-eslint/docs/packages/type-utils/generated/typedoc-sidebar.cjs
        0:0  error  Parsing error: /Users/josh/repos/typescript-eslint/docs/packages/type-utils/generated/typedoc-sidebar.cjs was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject
      
      ✖ 2 problems (2 errors, 0 warnings)
```

`packages/ast-spec/generated` and `packages/type-utils/generated` should be excluded from ESLint.

### Additional Info

💖
